pkg/virtctl/pause/pause.go

```diff
@@ -124,63 +97,45 @@ func (vc *virtCommand) Run(args []string) error {
-			if err != nil {
-				if errors.IsNotFound(err) {
-					runningStrategy, err := vm.RunStrategy()
-					if err != nil {
-						return fmt.Errorf("Error pausing VirtualMachineInstance %s: %v", vmiName, err)
-					}
-					if runningStrategy == kubevirtV1.RunStrategyHalted {
-						return fmt.Errorf("Error pausing VirtualMachineInstance %s. VirtualMachine %s is not set to run", vmiName, vm.Name)
-					}
-					return fmt.Errorf("Error pausing VirtualMachineInstance %s, it was not found", vmiName)
 
-				}
-				return fmt.Errorf("Error pausing VirtualMachineInstance %s: %v", vmiName, err)
-			}
-			printLog(vmiName, vc.command)
+	return executePauseCMD(virtClient, namespace, resourceType, resourceName, dryRunOption)
+}
 
-		case ARG_VMI_LONG, ARG_VMI_SHORT:
-			err = virtClient.VirtualMachineInstance(namespace).Pause(context.Background(), resourceName, &kubevirtV1.PauseOptions{DryRun: dryRunOption})
-			if err != nil {
-				return fmt.Errorf("Error pausing VirtualMachineInstance %s: %v", resourceName, err)
-			}
-			printLog(resourceName, vc.command)
+func executePauseCMD(client kubecli.KubevirtClient, namespace, resourceType, resourceName string, dryRunOption []string) error {
+	switch resourceType {
+	case ARG_VM_LONG, ARG_VM_SHORT:
+		vm, err := client.VirtualMachine(namespace).Get(context.Background(), resourceName, v1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("Error getting VirtualMachine %s: %v", resourceName, err)
 		}
-	case COMMAND_UNPAUSE:
-		switch resourceType {
-		case ARG_VM_LONG, ARG_VM_SHORT:
-			vm, err := virtClient.VirtualMachine(namespace).Get(context.Background(), resourceName, v1.GetOptions{})
-			if err != nil {
-				return fmt.Errorf("Error getting VirtualMachine %s: %v", resourceName, err)
-			}
-			vmiName := vm.Name
-			err = virtClient.VirtualMachineInstance(namespace).Unpause(context.Background(), vmiName, &kubevirtV1.UnpauseOptions{DryRun: dryRunOption})
-			if err != nil {
-				return fmt.Errorf("Error unpausing VirtualMachineInstance %s: %v", vmiName, err)
-			}
-			printLog(vmiName, vc.command)
-		case ARG_VMI_LONG, ARG_VMI_SHORT:
-			err = virtClient.VirtualMachineInstance(namespace).Unpause(context.Background(), resourceName, &kubevirtV1.UnpauseOptions{DryRun: dryRunOption})
-			if err != nil {
-				return fmt.Errorf("Error unpausing VirtualMachineInstance %s: %v", resourceName, err)
+		vmiName := vm.Name
+		err = client.VirtualMachineInstance(namespace).Pause(context.Background(), vmiName, &kubevirtV1.PauseOptions{DryRun: dryRunOption})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				runningStrategy, err := vm.RunStrategy()
+				if err != nil {
+					return fmt.Errorf("Error pausing VirtualMachineInstance %s: %v", vmiName, err)
+				}
+				if runningStrategy == kubevirtV1.RunStrategyHalted {
+					return fmt.Errorf("Error pausing VirtualMachineInstance %s. VirtualMachine %s is not set to run", vmiName, vm.Name)
+				}
```

Comment:
This validation should be covered by the API already. We shouldn't be doing any additional validations in virtctl, the API knows best.

If you drop this validation the code paths for VM/VMI could be unified and we're left with only a call to `client.VirtualMachineInstance(namespace).Pause`, which would be fine to do from `Run` and dropping this func.

Reply:
can we please limit this PR scope to the splitting ?
i must work on other stuff and cant dedicate much for it atm